rs/bitcoin/kyt/src/state.rs

```diff
@@ -122,3 +127,82 @@ impl Drop for FetchGuard {
         }
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    pub network: Network,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            network: Network::Bitcoin,
+        }
+    }
+}
+
+impl From<InitArg> for Config {
+    fn from(arg: InitArg) -> Self {
+        let network = match arg.network {
+            BtcNetwork::Mainnet => Network::Bitcoin,
+            BtcNetwork::Testnet => Network::Testnet,
+            BtcNetwork::Regtest => Network::Regtest,
+        };
+        Self { network }
+    }
+}
+
+impl From<&Config> for InitArg {
+    fn from(arg: &Config) -> Self {
+        let network = match arg.network {
+            Network::Bitcoin => BtcNetwork::Mainnet,
+            Network::Testnet => BtcNetwork::Testnet,
+            Network::Regtest => BtcNetwork::Regtest,
+            _ => panic!("unsupported network type: Signet"),
+        };
+        Self { network }
+    }
+}
+
+impl Storable for Config {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        let buf = encode_args((InitArg::from(self),)).expect("fail to encode config");
+        Cow::Owned(buf)
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        let (config,): (InitArg,) =
+            decode_args(bytes.as_ref()).expect("failed to decode config bytes");
+        config.into()
+    }
+
+    const BOUND: Bound = Bound::Unbounded;
+}
```

Comment:
I don't understand why you would want to store the state as an `InitArg` instead of a `Config` directly. I think storing it as `InitArg` is not advisable because you for example may want to have fields in the state that can only be changed via upgrade args.